controllers/cohortdata.go

```diff
@@ -21,6 +21,56 @@ func NewCohortDataController(cohortDataModel models.CohortDataI) CohortDataContr
 	return CohortDataController{cohortDataModel: cohortDataModel}
 }
 
+func (u CohortDataController) RetrieveHistogramForCohortIdAndConceptId(c *gin.Context) {
+	sourceIdStr := c.Param("sourceid")
+	log.Printf("Querying source: %s", sourceIdStr)
+	cohortIdStr := c.Param("cohortid")
+	log.Printf("Querying cohort for cohort definition id: %s", cohortIdStr)
+	if sourceIdStr == "" || cohortIdStr == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "bad request"})
+		c.Abort()
+		return
+	}
+
+	conceptIds, cohortPairs, err := utils.ParseConceptIdsAndDichotomousDefs(c)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "Error parsing request body for prefixed concept ids", "error": err.Error()})
+		c.Abort()
+		return
+	}
+
+	if len(conceptIds) == 0 {
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "must provide one or more concept ids for histogram endpoint"})
+		c.Abort()
+		return
+	}
+
+	// IMPORTANT ASSUMPTION: histogram concept_id is the last concept_id that we parse
+	histogramConceptId := conceptIds[len(conceptIds)-1]
```

Comment:
I see... let's add this `histogramConceptId` as a separate concept id to the endpoint